core/tools.js

```diff
@@ -962,8 +964,20 @@
 				}
 
 				if ( !( /%$/ ).test( cssLength ) ) {
+					var isNegative = parseFloat( cssLength ) < 0,
+						ret;
+
+					if ( isNegative ) {
+						cssLength = cssLength.replace( '-', '' );
+					}
+
 					calculator.setStyle( 'width', cssLength );
-					return calculator.$.clientWidth;
+					ret = calculator.$.clientWidth;
+
+					if ( isNegative ) {
```

Comment:
This logic could be a little bit simplified by using the technique similar to [`Math.sign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sign) (unfortunately it isn't supported in IE, so we can't use it directly).

However I'm not sure if it's really more readable or better in any way.

Reply:
>However I'm not sure if it's really more readable or better in any way.

Therefore I'll leave it as it is.